src/components/exhibitions/exhibition-content-rooms-screen.tsx

```diff
@@ -0,0 +1,205 @@
+import * as React from "react";
+
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { ReduxActions, ReduxState } from "../../store";
+
+import { History } from "history";
+import styles from "../../styles/exhibition-view";
+import { WithStyles, withStyles, CircularProgress } from "@material-ui/core";
+import { KeycloakInstance } from "keycloak-js";
+// eslint-disable-next-line max-len
+import { Exhibition, ExhibitionRoom } from "../../generated/client";
+import { AccessToken, CardMenuOption, BreadcrumbData } from '../../types';
+import Api from "../../api/api";
+import strings from "../../localization/strings";
+import CardList from "../generic/card/card-list";
+import CardItem from "../generic/card/card-item";
+import BasicLayout from "../generic/basic-layout";
+
+/**
+ * Component props
+ */
+interface Props extends WithStyles<typeof styles> {
+  history: History;
+  keycloak: KeycloakInstance;
+  accessToken: AccessToken;
+  exhibitionId: string;
+}
+
+/**
+ * Component state
+ */
+interface State {
+  loading: boolean;
+  exhibition?: Exhibition;
+  rooms: ExhibitionRoom[];
+}
+
+/**
+ * Component for exhibition content rooms view
+ */
+class ExhibitionContentRoomsScreen extends React.Component<Props, State> {
+
+  /**
+   * Constructor
+   *
+   * @param props component properties
+   */
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      loading: true,
+      rooms: []
+    };
+  }
+
+  /**
+   * Component did mount life cycle handler
+   */
+  public componentDidMount = async () => {
+    await this.fetchData();
+    this.setState({ loading: false });
```

Comment:
use same context for starting and stopping loading (start loading in the beginning of the method)